src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -194,47 +217,29 @@ private Main() {
      * @noinspection CallToPrintStackTrace, CallToSystemExit
      **/
     public static void main(String... args) throws IOException {
-        int errorCounter = 0;
-        boolean cliViolations = false;
+
+        final Main main = new Main();
+        final CommandLine commandLine = new CommandLine(main);
```

Comment:
Will do.

Reply:
An instance is needed because the annotated fields are injected with the command line option values. This can be an inner class. I will make the change.

Reply:
@remkop For your util, you pass in the instance. When would your `CommandLine` work with a non-default constructor instance? To me, it would seem the majority of main classes would use the default constructor.
You could have `CommandLine` accept `Class<?>` as well and just call `clss.newInstance()`.

Reply:
When parsing the command line, picocli will convert the command line options to the type of the annotated fields and inject the values into the fields. This is one of the advantages of using picocli vs. Commons CLI: usually you don’t need to inspect a `ParseResult`, you just use your annotated “bean” directly. 

So, we need the instance to get the parse results. For example:

```
class CliOptions {
  @Option(names = “-x”)
  int x;
}

CliOptions options = new CliOptions();
CommandLine cmd = new CommandLine(options);
String[] args = { “-x”, “123” };
cmd.parseArgs(args);
assert options.x == 123; // use instance
```

This is a common usage pattern. There are variations where you don’t need an instance, for example if the class implements `Runnable` or `Callable`. (And to support such use cases it is already possible to pass a `Class` to the `CommandLine` constructor and picocli will internally create an instance just like you describe.)

However, this application has requirements for error handling that means it’s simpler to just use the “common usage pattern” instead of the `Runnable` / `Callable` convenience methods.

Reply:
Done